labs/saw/arithmetic-verification/arithmetic_implementations.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+// addition_standard
+//
+// Add two 16-bit integers in a standard way, cast each to a local 32 bit
+// number and then add and return the result.
+uint32_t add_standard(uint16_t a, uint16_t b) {
+  uint32_t local_a = (uint32_t) a;
+  uint32_t local_b = (uint32_t) b;
+  uint32_t result;
+  
+  result = local_a + local_b;
+  
+  return result;
+}
+
+// addition_textbook
+//
+// Add two 16-bit numbers by treating each as two 8-bit parts and using
+// the "textbook" addition algorithm.
+//
+// Suppose we break up 16-bit integers a and be as follows: 
+//     a = [ a_1 ] [ a_0 ] 
+//     b = [ b_1 ] [ b_0 ]
+//
+// Then we compute a + b as:
+//                                         [ C_1 ]
+//                                         [ a_1 ] [ a_0 ]
+//                                       + [ b_1 ] [ b_0 ]
+//                                   =====================
+//                                 [ O_2 ] [ O_1 ] [ O_0 ]
+//
```

Comment:
Reader can probably suss what `C_1` means, but for the sake of completeness (and the fact that it's calculated in the code), it should probably be explicitly described.